File: CSqlManager/CSqlManager/Entities/ECU.cs
namespace CSqlManager;

public class ECU
{
    public string Brand_code { get; set; }
    public string code { get; set; }
    public string Fuel { get; set; }
    public bool dpf { get; set; }
    public bool egr { get; set; }
    public bool lambda { get; set; }
    public bool hotstart { get; set; }
    public bool flap { get; set; }
    public bool adblue { get; set; }
    public bool dtc { get; set; }
    public bool torqmonitor { get; set; }
    public bool speedlimit { get; set; }
    public bool startstop { get; set; }
    public bool nox { get; set; }
    public bool tva { get; set; }
    public bool readiness { get; set; }
    public bool immo { get; set; }
    public bool maf { get; set; }
    public bool hardcut { get; set; }
    public bool displaycalibration { get; set; }
    public bool waterpump { get; set; }
    public bool tprot { get; set; }
    public bool o2 { get; set; }
    public bool glowplugs { get; set; }
    public bool y75 { get; set; }
    public bool special { get; set; }
    public bool decata { get; set; }
    public bool vmax { get; set; }
    public bool stage1 { get; set; }
    public bool stage2 { get; set; }
    public bool flexfuel { get; set; }


    public ECU() { }

    public ECU(string brandCode, string ecuCode, bool[] toToggle)
    {
        Brand_code = brandCode;
        code = ecuCode;
        Fuel = "X";

        dpf = toToggle[0];
        egr =  toToggle[1];

        lambda = toToggle[2];
        hotstart =  toToggle[3];
        flap =  toToggle[4];
        dtc =  toToggle[5];
        adblue =  toToggle[6];
        torqmonitor = toToggle[7];

        speedlimit =  toToggle[8];
        startstop =  toToggle[9];
        nox =  toToggle[10];
        tva =  toToggle[11];
        readiness =  toToggle[12];
        immo = toToggle[13];

        maf = toToggle[14];
        hardcut = toToggle[15];
        displaycalibration = toToggle[16];
        waterpump = toToggle[17];
        tprot =  toToggle[18];

        o2 = toToggle[19];
        glowplugs = toToggle[20];
        y75 = toToggle[21];
        special =  toToggle[22];
        decata =  toToggle[23];
        vmax = toToggle[24];
        stage1 = toToggle[25];

        stage2 =  toToggle[26];
        flexfuel = toToggle[27];
        
    }
    
    public ECU(string brandCode, string ecuCode, string fuel, bool[] toToggle)
    {
        Brand_code = brandCode;
        code = ecuCode;
        Fuel = fuel;
        
        dpf = toToggle[0];
        egr =  toToggle[1];

        lambda = toToggle[2];
        hotstart =  toToggle[3];
        flap =  toToggle[4];
        dtc =  toToggle[5];
        adblue =  toToggle[6];
        torqmonitor = toToggle[7];

        speedlimit =  toToggle[8];
        startstop =  toToggle[9];
        nox =  toToggle[10];
        tva =  toToggle[11];
        readiness =  toToggle[12];
        immo = toToggle[13];

        maf = toToggle[14];
        hardcut = toToggle[15];
        displaycalibration = toToggle[16];
        waterpump = toToggle[17];
        tprot =  toToggle[18];

        o2 = toToggle[19];
        glowplugs = toToggle[20];
        y75 = toToggle[21];
        special =  toToggle[22];
        decata =  toToggle[23];
        vmax = toToggle[24];
        stage1 = toToggle[25];

        stage2 =  toToggle[26];
        flexfuel = toToggle[27];
       
    }

    public override string ToString()
    {
        return $"{Brand_code}|{code}";
    }
}

File: front/front/src/app/model/ecu.ts
export class Ecu {

    constructor(data: any) {
        this.code = data.code;
        this.brand_code = data.brand_code;
        this.fuel = data.fuel;
        this.dpf= data.dpf;
        this.egr= data.egr;
        this.lambda= data.lambda;
        this.hotstart= data.hotstart;
        this.flap= data.flap;
        this.adblue= data.adblue;
        this.dtc= data.dtc;
        this.torqmonitor= data.torqmonitor;
        this.speedlimit= data.speedlimit;
        this.startstop= data.startstop;
        this.nox= data.nox;
        this.tva= data.tva;
        this.readiness= data.readiness;
        this.immo= data.immo;
        this.maf= data.maf;
        this.hardcut= data.hardcut;
        this.displaycalibration= data.displaycalibration;
        this.waterpump= data.waterpump;
        this.tprot= data.tprot;
        this.o2= data.o2;
        this.glowplugs= data.glowplugs;
        this.y75= data.y75;
        this.special= data.special;
        this.decata= data.decata;
        this.vmax= data.vmax;
        this.stage1= data.stage1;
        this.stage2= data.stage2;
        this.flexfuel= data.flexfuel;

    }

    brand_code: string;
    code: string;
    fuel: string;

    dpf: boolean;
    egr: boolean;
    lambda: boolean;
    hotstart: boolean;
    flap: boolean;
    adblue: boolean;
    dtc: boolean;
    torqmonitor: boolean;
    speedlimit: boolean;
    startstop: boolean;
    nox: boolean;
    tva: boolean;
    readiness: boolean;
    immo: boolean;
    maf: boolean;
    hardcut: boolean;
    displaycalibration: boolean;
    waterpump: boolean;
    tprot: boolean;
    o2: boolean;
    glowplugs: boolean;
    y75: boolean;
    special: boolean;
    decata: boolean;
    vmax: boolean;
    stage1: boolean;
    stage2: boolean;
    flexfuel: boolean;

    processing: string = '';

    
  getProcessing(): string {
    if (this.processing != '') {
      return this.processing;
    }
    let result: string[] = [];
    if (this.dpf) { result.push('dpf');};
    if (this.egr) { result.push('egr');};
    if (this.lambda) { result.push('lambda');};
    if (this.hotstart) { result.push('hotstart');};
    if (this.flap) { result.push('flap');};
    if (this.adblue) { result.push('adblue');};
    if (this.dtc) { result.push('dtc');};
    if (this.torqmonitor) { result.push('torqmonitor');};
    if (this.speedlimit) { result.push('speedlimit');};
    if (this.startstop) { result.push('startstop');};
    if (this.nox) { result.push('nox');};
    if (this.tva) { result.push('tva');};
    if (this.readiness) { result.push('readiness');};
    if (this.immo) { result.push('immo');};
    if (this.maf) { result.push('maf');};
    if (this.hardcut) { result.push('hardcut');};
    if (this.displaycalibration) { result.push('displaycalibration');};
    if (this.waterpump) { result.push('waterpump');};
    if (this.tprot) { result.push('tprot');};
    if (this.o2) { result.push('o2');};
    if (this.glowplugs) { result.push('glowplugs');};
    if (this.y75) { result.push('y75');};
    if (this.special) { result.push('special');};
    if (this.decata) { result.push('decata');};
    if (this.vmax) { result.push('vmax');};
    if (this.stage1) { result.push('stage1');};
    if (this.stage2) { result.push('stage2');};
    if (this.flexfuel) { result.push('flexfuel');};

    let s = '';
    for(const p of result) {
      if (s.length > 0) {
        s = s + ', ';
      }
      s = s + p;
    }
    this.processing = s;
    return this.processing;
  }
}